### PR TITLE
Fixed wire-compiler tests. They were failing on Windows.

### PR DESCRIPTION
**Pull request by davidpelfree**
I am using Windows, and there were a few problems in tests. I fixed the following on `wire-compiler` project, mainly on `WireCompilerTest` class:

1.  Relying on "current working directory" (`new File(".")`) was not working correctly on Windows. Instead I relied on searching classpath for "target/test-classes" directory and take its parent.

2. Temp directory moved under "target" in order to be cleaned automatically by Maven "clean" task. This is not related to using Windows, it's a general better idea.

3. Scanner instances were not closed. This prevented `cleanup()` from working correctly on Windows - files could not be deleted, since they were still open.

I would greatly appreciate if you could review my fixes and merge them into your code. 

David.

### CODE REVIEW

**Comment by JakeWharton** [comment]
Needs a rebase.

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java [hunk 1]
```diff
@@ -20,26 +20,51 @@
...
 
   private File testDir;
+  private File wireRuntimeDir;
+
+  @Before
+  public void setUp() throws IOException {
+    String targetDir = locateProjectTargetDirectory();
+
+    // Create temp dir under "target" so it would clean during Maven "clean" task.
+    testDir = makeTestDirectory(targetDir + "WireCompilerTest");
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java]: Use a `TemporaryFolder` JUnit `@Rule` instead.

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java [hunk 2]
```diff
@@ -625,9 +650,9 @@ private void cleanupHelper(File f) {
           cleanupHelper(ff);
         }
       }
-      f.delete();
-    } else {
-      f.delete();
+    }
+    if (!f.delete()) {
+        System.err.println("Cannot delete: " + f);
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java]: nit: two space indent

File: wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java [hunk 3]
```diff
@@ -20,26 +20,51 @@
 import java.io.File;
 import java.io.FileNotFoundException;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Scanner;
-import java.util.Set;
+import java.net.URL;
+import java.util.*;
```
Comment (JakeWharton) [inline wire-compiler/src/test/java/com/squareup/wire/WireCompilerTest.java]: Undo star import

**Comment by danrice-square** [comment]
@davidpelfree I'll take a look at this later today. Quick question: have you signed a Square contributor agreement?

**Comment by davidpelfree** [comment]
@danrice-square - yes.

**Comment by danrice-square** [comment]
LGTM once Jake's comments are addressed.

**Comment by davidpelfree** [comment]
Guys, I spent my time making these changes in order to make Wire a better product. 
I am not sure if I understand your last comment correctly but you're invited to take my contribution and make your changes and integrate it. 
The ball is now on your field.

**Comment by danrice-square** [comment]
Hi David - there are a few changes that Jake requested. Since we don't develop on Windows, it would be best if you make those changes, ensure the code passes tests, then push the changes onto your branch. Then we will merge the code.

"LGTM" means "looks good to me." It means I am willing to merge the code one the issues that have been raised in comments have all been addressed.

**Comment by davidpelfree** [comment]
Hi @danrice-square. Thanks for the clarification. I will see if I can resolve these comments soon.

**Comment by swankjesse** [comment]
Sorry for dropping the ball on this. I'm closing as obsolete!

**Pull request closed without merging**